core/common/src/main/scala/org/typelevel/otel4s/TextMapInjector.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+trait TextMapInjector[A] {
+  type Builder
+
+  def textMapSetter: TextMapSetter[Builder]
+
+  def toBuilder(carrier: A): Builder
+
+  def toCarrier(builder: Builder): A
+}
```

Comment:
I'm not a fan of this API. it requires much more effort to implement than `TextMapSetter`, to the point that you may need to implement `TextMapSetter` *on top* of it. personally I think that the builder and iteration over it should be handled internally. if performance is essential, they can use a `TextMapSetter` directly anyway

Reply:
oh neat. `TextMapUpdater` is exactly the API I was thinking of

Reply:
is there an advantage to `val box: Box[A]` over `var updatedCarrier: A`?

Reply:
We probably don't even need the box.  Since we're ignoring the first parameter to the Java TextMapSetter, some dummy singleton (or even null, but ew) should work.